src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDocletWriter.java

```diff
@@ -1543,7 +1544,8 @@ private boolean inSamePackage(Element element) {
         Element currentPageElement = (this instanceof PackageWriterImpl packageWriter)
                 ? packageWriter.packageElement : getCurrentPageElement();
         return currentPageElement != null && !utils.isModule(element)
-                && utils.containingPackage(currentPageElement) == utils.containingPackage(element);
+                && Objects.equals(utils.containingPackage(currentPageElement),
```

Comment:
I would expect that that containing package is never null, so it should be OK to use simple `.equals`, not `Objects.equals`.

Hmm, OK, the package of a module is `null`.